interop-test-descriptions.md

```diff
@@ -0,0 +1,82 @@
+gRPC-Web Interop Tests
+======================
+
+This document describes the set of tests any gRPC-Web clients or proxies need
+to implement. The proto definition for the messages and RPCs we are using for
+the tests can be found
+[here](src/proto/grpc/testing/test.proto).
+
+The canonical set of interop tests was defined in the main
+[grpc/grpc repo](https://github.com/grpc/grpc/blob/master/doc/interop-test-descriptions.md).
+
+Here in gRPC-Web, we will only implement a subset of tests that's relevant to
+gRPC-Web. For example, we will not be implementing any tests involving
+client-streaming or bidi-streaming for now. On the other hand, there are
+gRPC-Web specific tests that we will add here.
+
+```
+ gRPC-Web Client  <-->  Proxy  <--> gRPC Service
+```
+
+The idea is that we should be able to swap out any of the 3 components above
+and all the interop tests should still pass.
+
+This repository will provide a canonical implementation of the interop test
+suite using the Javascript client, Envoy and a gRPC service implemented in
+Node.
+
+For any new gRPC-Web client implementation, you need to swap out the JS
+client and make sure all tests still pass.
+
+For any in-process proxies implementation, you need to swap out the proxy
+and the service as a unit and make sure the standard JS client will still
+pass the tests.
+
+
+List of Tests
+-------------
+
+| Test Name | grpc-web-text Mode | grpc-web Binary mode |
+| --------- |:------------------:|:--------------------:|
+| empty_unary | &#10003; | &#10003; |
+| cacheable_unary | TBD | TBD |
+| large_unary | &#10003; | &#10003; |
+| client_compressed_unary | TBD | TBD |
+| server_compressed_unary | TBD | TBD |
+| client_streaming | &#10007; | &#10007; |
+| client_compressed_streaming | &#10007; | &#10007; |
+| server_streaming | &#10003; | &#10007; |
+| server_compressed_streaming | TBD | &#10007; |
+| ping_pong | &#10007; | &#10007; |
+| empty_stream | &#10007; | &#10007; |
+| compute_engine_creds | TBD | TBD |
+| jwt_token_creds | TBD | TBD |
+| oauth2_auth_token | TBD | TBD |
+| per_rpc_creds | TBD | TBD |
+| google_default_credentials | TBD | TBD |
+| compute_engine_channel_credentials | TBD | TBD |
+| custom_metadata | &#10003; | &#10003; |
+| status_code_and_message | &#10003; | &#10003; |
```

Comment:
status_code_and_message has a full duplex call.

https://github.com/grpc/grpc/blob/f12df974aeb305d3fc322b2b3aaab38d46b0c30a/doc/interop-test-descriptions.md#status_code_and_message

Reply:
Yea I probably should put an asterisk in those tests where we should only implement the `UnaryCall` part.

Reply:
I think this is the only one like that with a tick.